server/sonar-server/src/main/java/org/sonar/server/measure/custom/ws/CreateAction.java

```diff
@@ -0,0 +1,225 @@
+ */
+
+package org.sonar.server.measure.custom.ws;
+
+import com.google.common.base.Strings;
+import java.net.HttpURLConnection;
+import org.sonar.api.measures.Metric;
+import org.sonar.api.server.ws.Request;
+import org.sonar.api.server.ws.Response;
+import org.sonar.api.server.ws.WebService;
+import org.sonar.api.utils.System2;
+import org.sonar.api.utils.text.JsonWriter;
+import org.sonar.api.web.UserRole;
+import org.sonar.core.component.ComponentDto;
+import org.sonar.core.measure.custom.db.CustomMeasureDto;
+import org.sonar.core.metric.db.MetricDto;
+import org.sonar.core.permission.GlobalPermissions;
+import org.sonar.core.persistence.DbSession;
+import org.sonar.core.persistence.MyBatis;
+import org.sonar.server.db.DbClient;
+import org.sonar.server.exceptions.NotFoundException;
+import org.sonar.server.exceptions.ServerException;
+import org.sonar.server.user.UserSession;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+public class CreateAction implements CustomMeasuresWsAction {
+  public static final String ACTION = "create";
+  public static final String PARAM_PROJECT_ID = "projectId";
+  public static final String PARAM_PROJECT_KEY = "projectKey";
+  public static final String PARAM_METRIC_ID = "metricId";
+  public static final String PARAM_VALUE = "value";
+  public static final String PARAM_DESCRIPTION = "description";
+
+  private static final String FIELD_ID = "id";
+  private static final String FIELD_PROJECT_ID = PARAM_PROJECT_ID;
+  private static final String FIELD_PROJECT_KEY = PARAM_PROJECT_KEY;
+  private static final String FIELD_METRIC_ID = PARAM_METRIC_ID;
+  private static final String FIELD_VALUE = PARAM_VALUE;
+  private static final String FIELD_DESCRIPTION = PARAM_DESCRIPTION;
+
+  private final DbClient dbClient;
+  private final UserSession userSession;
+  private final System2 system;
+
+  public CreateAction(DbClient dbClient, UserSession userSession, System2 system) {
+    this.dbClient = dbClient;
+    this.userSession = userSession;
+    this.system = system;
+  }
+
+  @Override
+  public void define(WebService.NewController context) {
+    WebService.NewAction action = context.createAction(ACTION)
+      .setDescription("Create a custom measure.<br /> " +
+        "The project id or the project key must be provided. <br/>" +
+        "Requires 'Administer System' permission or 'Administer' permission on the project.")
+      .setSince("5.2")
+      .setPost(true)
+      .setHandler(this);
+
+    action.createParam(PARAM_PROJECT_ID)
+      .setDescription("Project id")
+      .setExampleValue("ce4c03d6-430f-40a9-b777-ad877c00aa4d");
+
+    action.createParam(PARAM_PROJECT_KEY)
+      .setDescription("Project key")
+      .setExampleValue("org.apache.hbas:hbase");
+
+    action.createParam(PARAM_METRIC_ID)
+      .setRequired(true)
+      .setDescription("Metric id")
+      .setExampleValue("16");
+
+    action.createParam(PARAM_VALUE)
```

Comment:
This parameter could have a description of the supported formats. Especially for BOOL (since the WS convention allows ``true``, ``false``, ``yes``, ``no``) and WORK_DUR.